sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -54,4 +128,31 @@ class RocksDBSuite extends SparkFunSuite {
       """{"sstFiles":[{"localFileName":"00001.sst","dfsSstFileName":"00001-uuid.sst","sizeBytes":12345678901234}],"logFiles":[{"localFileName":"00001.log","dfsLogFileName":"00001-uuid.log","sizeBytes":12345678901234}],"numKeys":12345678901234}""")
     // scalastyle:on line.size.limit
   }
+
+
+  def generateFiles(dir: String, fileToLengths: Seq[(String, Int)]): Unit = {
+    fileToLengths.foreach { case (fileName, length) =>
+      val file = new File(dir, fileName)
+      FileUtils.write(file, "a" * length)
+    }
+  }
+
+  def saveCheckpointFiles(
+      fileManager: RocksDBFileManager,
+      fileToLengths: Seq[(String, Int)],
+      version: Int,
+      numKeys: Int): Unit = {
+    val checkpointDir = Utils.createTempDir().getAbsolutePath // local dir to create checkpoints
+    generateFiles(checkpointDir, fileToLengths)
+    fileManager.saveCheckpointToDfs(checkpointDir, version, numKeys)
+  }
+
+  implicit def toFile(path: String): File = new File(path)
```

Comment:
With this, `def listFiles(file: String)` is probably unnecessary, as I expect implicit conversion happening when calling `listFiles(file: File)` with String.

Reply:
That's right in this PR. In the further PRs, we will have more caller for the `listFiles(file: String)`. Maybe let's keep it for now.